spark/src/main/resources/python/zeppelin_pyspark.py

```diff
@@ -44,15 +49,59 @@ def flush(self):
 class PyZeppelinContext(dict):
   def __init__(self, zc):
     self.z = zc
+    self.max_result = 1000
     self._displayhook = lambda *args: None
 
+  def _isPandas(self, obj):
+    try:
+      from pandas.core.frame import DataFrame
+      return isinstance(obj, DataFrame)
+    except ImportError:
+      return false
+
   def show(self, obj):
     from pyspark.sql import DataFrame
     if isinstance(obj, DataFrame):
       print(self.z.showData(obj._jdf))
+    elif self._isPandas(obj):
+      self.show_dataframe(obj)
     else:
       print(str(obj))
 
+  def show_dataframe(self, df, show_index=False):
+    """Pretty prints DF using Table Display System
```

Comment:
how much of this overlap with `z.showData`? should we abstract out a method in `zeppelin context` to do pretty print